node/reqres.js

```diff
@@ -284,8 +287,13 @@ TChannelOutgoingResponse.prototype.sendError = function sendError(codeString, me
     if (self.state === States.Done || self.state === States.Error) {
         throw new Error('response already done'); // TODO: typed error
     } else {
-        self.sendFrame.error(codeString, message);
+        // TODO: we could decide to flush any parts in a (first?) call res frame
+        self.finished = true;
         self.state = States.Error;
+        if (!self.arg1._readableState.ended) self.arg1.end();
```

Comment:
we should check _writableState instead `end()` is a method on writable.

Apparently double end() ( https://github.com/joyent/node/blob/v0.10.32-release/lib/_stream_writable.js#L342-L343 ) is idempotent.